lib/ansible/galaxy/collection.py

```diff
@@ -410,27 +471,25 @@ def publish_collection(collection_path, api, wait, timeout):
     display.display("Publishing collection artifact '%s' to %s %s" % (collection_path, api.name, api.api_server))
 
     n_url = _urljoin(api.api_server, 'api', 'v2', 'collections')
+    available_api_versions = get_available_api_versions(api)
```

Comment:
Ditto here, available_versions should be a property on the api instance which is lazily collected as needed.